docs/source/troubleshooting.md

```diff
@@ -0,0 +1,189 @@
+
+In order to be capable of automatically [upgrading](https://github.com/hyperledger/indy-node/blob/master/docs/source/pool-upgrade.md) itself Indy Node employs separate service called `indy-node-control`, which runs along with main service called `indy-node`.
+Also it is worth noting that `indy-node` service is configured to automatically restart node process in case it crashes.
+
+### Types of failures
+
+In an ideal world network connections are always stable, nodes do not fail, and software doesn't have bugs. 
+Unfortunatelly in real world this is not the case, so Indy Node pool can fail in some circumstances.
+
+Most of failures can divided into following categories (in order of increasing severity):
+- failures induced by environment problems, like misconfigured firewalls preventing nodes connecting to each others
+- transient consensus failures on some nodes, most likely to some unhandled edge cases, which can go away after restarting affected nodes
+- node failures, like inability to properly perform an upgrade or handle some incoming request due to some bug, which require manual intervention, but doesn't affect ledger data
+- ledger corruption, which require touching ledger data in order to fix it
+
+Of course, if number of affected nodes is _f_ or less then from external point of view functionality of pool will be unaffected.
+However if more than _f_ nodes become affected then pool will not be able to do writes, although pool still will be able to process reads until all nodes fail.
+
+### Where to get info
+
+Most useful places get info are the following:
+- [Indy CLI](https://github.com/hyperledger/indy-sdk/tree/master/cli) can be used for sending read and write requests to pool, as well as checking general connectivity.
+- Either VALIDATOR_INFO command sent through Indy CLI (in case of Sovrin network you'll need to have keys for priveleged DID to do so), or [validator-info](https://github.com/hyperledger/indy-node/blob/master/design/validator_info.md) script run on validator node.
+  These tools provide important information like how many nodes are connected to each other, when last write happened (due to freshness check it should happen at least once per 5 minutes), whether a view change is in progress now or other important data, which is useful to assess pool current health and can be a starting point for further investigation, when needed.
+- `journalctl` logs can be useful because they contain tracebacks of indy-node crashes, if they happened, and these logs are really easy to check.
+  Sometime crashes can be due to some bugs in code, but also they can be caused by insufficient resource (either memory or disk space), and if this is the case `journalctl` logs can provide a quick answer.
+- Indy node logs, located in `/var/log/indy/<network name>/`.
```

Comment:
Also, please make sure, that we have logs, located on `/var/log/indy/node_control.log`. It can helps with debugging upgrade procedure, because it's logs from node-control-tool

Reply:
Done